buildsys.cmake

```diff
@@ -42,10 +42,10 @@ macro(cpp_msvc_overrides)
   add_compile_options($<$<COMPILE_LANGUAGE:C,CXX>:/Ot>)
 
   # disable security checks (stack buffer overrun prevention)
-  add_compile_options($<$<AND:$<COMPILE_LANGUAGE:C,CXX>,$<CONFIG:RELEASE>>:/GS->)
-  add_compile_options($<$<AND:$<COMPILE_LANGUAGE:C,CXX>,$<CONFIG:RELEASE>>:/O2>)
-  add_compile_options($<$<AND:$<COMPILE_LANGUAGE:C,CXX>,$<CONFIG:RELWITHDEBINFO>>:/GS->)
-  add_compile_options($<$<AND:$<COMPILE_LANGUAGE:C,CXX>,$<CONFIG:RELWITHDEBINFO>>:/O2>)
+  add_compile_options($<$<AND:$<COMPILE_LANGUAGE:C,CXX>,$<CONFIG:Release>>:/GS->)
```

Comment:
Don't we call `string(TOUPPER "${CMAKE_BUILD_TYPE}" BT)` before this?

Reply:
This does not change the CMAKE_BUILD_TYPE variable, and:

> CMAKE_BUILD_TYPE Specifies the build type on single-configuration generators.
https://cmake.org/cmake/help/latest/variable/CMAKE_BUILD_TYPE.html

Reply:
This does not change the CMAKE_BUILD_TYPE variable, and:

CMAKE_BUILD_TYPE Specifies the build type on single-configuration generators.
https://cmake.org/cmake/help/latest/variable/CMAKE_BUILD_TYPE.html